app/controllers/my_facilities/drug_stocks_controller.rb

```diff
@@ -59,9 +59,17 @@ def create_drug_report
       .includes(facility_group: :protocol_drugs)
       .where(protocol_drugs: {stock_tracked: true})
 
+    @blocks = Region
+      .block_regions
+      .joins("INNER JOIN regions facility_region ON regions.path @> facility_region.path")
+      .where(facility_region: {source_id: @facilities})
+      .distinct("regions.id")
```

Comment:
I've seen some elegant ways to fetch blocks for a list of regions somewhere else in our codebase I think. Wanna plop this in #server for some quick ideas?

Reply:
Since `@facilities` is a set of filtered facilities, we can't fetch blocks top down and we need to go from the facilities, bottom up. The elegant ways don't work for this. We tried a few different queries, out of which the current one is most performant:
```
Region.where(id: @facilities.with_block_region_id.pluck(:block_region_id).uniq)
```
```
Region.where(id:
 Region
   .where(source_id: @facilities.pluck(:id))
   .joins("INNER JOIN regions block_region ON block_region.path @> regions.path AND block_region.region_type = 'block'")
   .select("block_region.*")
   .distinct
)
```

Reply:
Cooooolbeans, you're way ahead of me 🚀